auslib/web/admin/swagger/api.yaml

```diff
@@ -259,21 +757,87 @@ definitions:
       - update_type
       - version
 
+  RulesListGET:
+    title: Rules List
+    description: Returns JSON object of list of rules with their count
+    type: object
+    required:
+      - count
+      - rules
+    properties:
+      count:
+        description: count of total rules fetched
+        type: integer
+        minimum: 0
+        format: int32
+        example: 1
+      rules:
+        description: array where each element is a rule object
+        type: array
+        items:
+          $ref: '#/definitions/RulesBaseGET'
+        example:
+          - alias: "esr45-localtest"
+            backgroundRate: 100
+            buildID: null
+            buildTarget: null
+            channel: "esr-localtest"
+            comment: "esr45 esr-localtest rule"
+            data_version: 22
+            distVersion: null
+            distribution: null
+            fallbackMapping: null
+            headerArchitecture: null
+            locale: null
+            mapping: "Firefox-45.7.0esr-build1"
+            osVersion: null
+            priority: 43
+            product: "Firefox"
+            rule_id: 308
+            systemCapabilities: null
+            update_type: "minor"
+            version: null
+            whitelist: null
+
+  UserRoles:
+    title: User Roles object definition
+    description: Users may hold any number of Roles. Roles are used when signing off on Scheduled Changes.
+    type: object
+    properties:
+      data_version:
+        description: the version number of user role
+        type: ["integer", "string"]
```

Comment:
same issue that occurred with all other integer fields such as priority and backgroundRate. The rule_id is passed into request body when duplicate existing rule form submits data.